K0s docs tw review (#854)
k0sproject/k0s: docs/k0sctl-install.md

# Install using k0sctl

k0sctl is a command-line tool for bootstrapping and managing k0s clusters. k0sctl connects to the provided hosts using SSH and gathers information on
the hosts, with which it forms a cluster by configuring the hosts, deploying k0s, and then connecting the k0s nodes together.

![k0sctl deployment](img/k0sctl_deployment.png)

> With k0sctl, you can create multi-node clusters for remote hosts in a manner that is automatic and easily repeatable. This method is recommended for production cluster installation.

Reviewer: "clusters for remote hosts" sounds wrong. Maybe drop that part altogether:

```suggestion
With k0sctl, you can create multi-node clusters in a manner that is automatic and easily repeatable. This method is recommended for production cluster installation.
```

(recommending k0sctl for production sounds a bit scary considering the stage of development it is in, but that is a different question)

> **Note**: The k0sctl install method is necessary for automatic upgrade.

Reviewer: Not sure where this came from or what it means. There is no automatic upgrade AFAIK. cc: @jnummelin

## Prerequisites

> k0sctl can be executed on Linux, MacOS, and Windows (for more detail, refer to the [k0sctl github repository](https://github.com/k0sproject/k0sctl)).

Reviewer: It can be executed on a bunch of other systems too. The releases page provides binaries for the listed platforms, but k0sctl should work just fine on any of the platforms supported by go (DragonFly BSD, FreeBSD, Linux, macOS, NetBSD, OpenBSD, Plan 9, Solaris, Windows) if you compile it yourself.

Anyways, executions are not fun, I would maybe change this to something like "You can find pre-compiled k0sctl binaries for Linux, macOS and Windows on the [k0sctl releases](https://...) page".

Author: Changed to read: "You can execute k0sctl on any system that supports the Go language. Pre-compiled k0sctl binaries are availble on the [k0sctl releases page](https://github.com/k0sproject/k0sctl))."

> **Note**: For hosts running k0s, refer to the [System Requirements](system-requirements.md).

Reviewer: ```suggestion
**Note**: For target hosts, refer to the [k0s System Requirements](system-requirements.md).
```
🤔

Author: The suggested note reads redundant ("For system requirement refer to system requirements."...?)

How about: **Note**: For target host prerequisites information, refer to the [k0s System Requirements](system-requirements.md).

## Install k0s

### 1. Install k0sctl tool

k0sctl is a single binary, the instructions for downloading and installing of

> which are available in the [k0sctl github repository](https://github.com/k0sproject/k0sctl#installation).

Reviewer: "installing of which are"

> # Install using k0sctl

Reviewer 2: Note that when updating page titles, you should also update the navigation accordingly. See the file mkdocs.yml in the parent directory.

Author: Thinking that I should update the mkdocs.yml only once a title change has been merged.

Reviewer 3: both changes has to be done simultaneously, otherwise the docs rendering gets "broken"

Author: OK. Understood.

INFO      k0sctl kubeconfig              
```

### 4. Access the cluster

> You must have the `kubeconfig` to access your k0s cluster.

Reviewer: Sounds a bit blunt, considering the section is about how you can do exactly that. Now it sounds like a prerequisite.

```suggestion
You can use k0sctl to generate a `kubeconfig` for accessing your k0s cluster.
```

Author: Editing to read: "To access your k0s cluster, use k0sctl to generate a `kubeconfig` for the purpose."

```sh
$ k0sctl kubeconfig > kubeconfig
```

With the `kubeconfig`, you can access your cluster using either kubectl or

> [LENS](https://k8slens.dev/).

Reviewer: LENS probably shouldn't be all caps

kube-system   metrics-server-6fbcd86f7b-5frtn            1/1     Running   0          3m51s
```

## Upgrade a k0s cluster using k0sctl

The upgrading of k0s clusters occurs not through a particular command (there is

> no ``upgrade`` sub-command in k0sctl) but by way of the configuration file. The

Reviewer: ```suggestion
no `upgrade` sub-command in k0sctl) but by way of the configuration file. The
```

configuration file describes the desired state of the cluster. and when that

> description passes to the `k0sctl apply` command a discovery of the current

Reviewer: it has to be passed, it doesn't pass on it's own :) 

Also something went wrong with the dot `.`.

Author: Now reads: "...cluster, and when you pass the description to the `k0sctl apply` command..."

state is performed and the system does whatever is necessary to bring the
cluster to the desired state (for example, perform an upgrade).

### k0sctl cluster upgrade process

The following operations occur during a k0sctl upgrade:

1. Upgrade of each controller, one at a time. There is no downtime if multiple

>    conrollers are configured.

Reviewer: ```suggestion
   controllers have been configured.
```

2. Upgrade of workers, in batches of 10%.

3. Draining of workers, which allows the workload to move to other nodes
   prior to the actual upgrade of the worker node components. (To skip the
   drain process, use the ``--no-drain`` option.)

4. The upgrade process continues once the upgraded nodes return to **Ready**
   state.

> You can onfigure the desired cluster version in the k0sctl configuration by

Reviewer: ```suggestion
You can configure the desired cluster version in the k0sctl configuration by
```
